facilitator/src/bin/facilitator.rs

```diff
@@ -1064,32 +1043,67 @@ fn aggregate_subcommand(sub_matches: &ArgMatches) -> Result<(), anyhow::Error> {
-    let identity = sub_matches.value_of("task-queue-identity");
-    let gcp_project_id = sub_matches.value_of("gcp-project-id").unwrap();
-    let pubsub_api_endpoint = sub_matches.value_of("pubsub-api-endpoint");
+    let aggregation_started: Counter = register_counter!(
+        "aggregation_jobs_begun",
+        "Number of aggregation jobs that started (on the facilitator side)"
+    )
+    .context("failed to register counter for started aggregations")?;
+    let aggregation_finished = register_counter_vec!(
+        "aggregation_jobs_finished",
+        "Number of aggregation jobs that finished (on the facilitator side)",
+        &["status"]
+    )
+    .context("failed to register counter for finished aggregations")?;
 
-    let mut queue: PubSubTaskQueue<AggregationTask> =
-        PubSubTaskQueue::new(pubsub_api_endpoint, gcp_project_id, queue_name, identity)?;
+    let mut queue = aggregation_task_queue_from_args(sub_matches)?;
 
     loop {
         if let Some(task_handle) = queue.dequeue()? {
             info!("dequeued task: {:?}", task_handle);
+            aggregation_started.inc();
 
             let batches: Vec<(&str, &str)> = task_handle
                 .task
                 .batches
                 .iter()
                 .map(|b| (b.id.as_str(), b.time.as_str()))
                 .collect();
-            aggregate(
+            let result = aggregate(
                 &task_handle.task.aggregation_id,
                 &task_handle.task.aggregation_start,
                 &task_handle.task.aggregation_end,
                 batches,
                 sub_matches,
-            )?;
-            queue.acknowledge_task(task_handle)?;
+            );
+
+            match result {
+                Ok(_) => {
+                    aggregation_finished.with_label_values(&["success"]).inc();
+                    queue.acknowledge_task(task_handle)?;
+                }
+                Err(err) => {
+                    error!("error while processing task {:?}: {:?}", task_handle, err);
+                    aggregation_finished.with_label_values(&["error"]).inc();
+                }
+            }
+
+            // Once we've handled teh task, whether it succeeds or fails, we
+            // want to push any metrics we have collected.
+            if let Some(pushgateway) = sub_matches.value_of("pushgateway") {
+                if pushgateway != "" {
+                    info!("pushing metrics to {}", pushgateway);
+                    prometheus::push_metrics(
+                        "aggregate",
+                        prometheus::labels! {},
+                        pushgateway,
+                        prometheus::gather(),
+                        None,
+                    )
+                    .map(|_| info!("done pushing metrics"))
+                    .map_err(|e| error!("error pushing metrics: {}", e))
+                    .ok();
+                }
+            }
```

Comment:
This block, which is duplicated between the intake-batch and aggregate workers, should be factored out into a function.

Reply:
Elsewhere we discussed moving the facilitator to a pull/scrape model for Prometheus metrics, so rather than refactor this and test it, I'm just going to delete it, since that's what the subsequent change to add a metrics HTTP endpoint would do anyway.